src/coreclr/src/vm/readytoruninfo.h

```diff
@@ -48,13 +74,18 @@ class ReadyToRunInfo
 
     PTR_PersistentInlineTrackingMapR2R m_pPersistentInlineTrackingMap;
 
-    ReadyToRunInfo(Module * pModule, PEImageLayout * pLayout, READYTORUN_HEADER * pHeader, AllocMemTracker *pamTracker);
-
 public:
+    ReadyToRunInfo(Module * pModule, PEImageLayout * pLayout, READYTORUN_HEADER * pHeader, DomainCompositeImage * compositeImage, AllocMemTracker *pamTracker);
+
     static BOOL IsReadyToRunEnabled();
 
     static PTR_ReadyToRunInfo Initialize(Module * pModule, AllocMemTracker *pamTracker);
 
+    bool IsComponentAssembly() const { return m_isComponentAssembly; }
+    const ReadyToRunCoreInfo * GetComponent() const { return &m_component; }
```

Comment:
A nit - it seems that GetComponentInfo / GetCompositeInfo or something along those lines would be a better to understand name.